src/text/raw_text_reader.rs

```diff
@@ -11,45 +9,43 @@ use crate::{Decimal, Integer, IonError, IonType};
 use crate::text::non_blocking::raw_text_reader::RawTextReader as NonBlockingReader;
 
 const INITIAL_PARENTS_CAPACITY: usize = 16;
-const READER_BUFFER_CAPACITY: usize = 1024 * 4;
+const READER_DEFAULT_BUFFER_CAPACITY: usize = 1024 * 4;
 
 pub struct RawTextReader<T: ToIonDataSource> {
     source: T::DataSource,
     // The inner non-blocking reader.
     reader: NonBlockingReader<Vec<u8>>,
     // The number of bytes fed to the inner reader.
     bytes_fed: usize,
-}
-
-/// Represents the final outcome of a [RawTextReader]'s attempt to parse the next value in the stream.
-///
-/// `IonParseResult<'a>` is not suitable for this purpose; its lifetime, `'a`, causes
-/// the result to hold a reference to the [RawTextReader]'s buffer even after parsing has finished,
-/// making it difficult to perform the necessary bookkeeping that follows finding the next value.
-///
-/// This type is intentionally limited in what it stores to avoid having a lifetime.
-#[derive(Eq, PartialEq, Debug)]
-pub(crate) enum RootParseResult<O> {
-    Ok(O),
-    Eof,
-    NoMatch,
-    Failure(String),
+    // The maximum amount of data to read at once.
+    max_read_size: usize,
 }
 
 impl<T: ToIonDataSource> RawTextReader<T> {
-    pub fn new(input: T) -> RawTextReader<T> {
-        let buffer = Vec::with_capacity(READER_BUFFER_CAPACITY);
-        RawTextReader {
+    pub fn new(input: T) -> IonResult<RawTextReader<T>> {
+        Self::new_with_size(input, READER_DEFAULT_BUFFER_CAPACITY)
+    }
+
+    pub fn new_with_size(input: T, size: usize) -> IonResult<RawTextReader<T>> {
+        let buffer = Vec::with_capacity(size);
+        let mut reader = RawTextReader {
             source: input.to_ion_data_source(),
             reader: NonBlockingReader::new(buffer),
             bytes_fed: 0,
-        }
+            max_read_size: size,
+        };
+        reader.read_source(size)?;
+        Ok(reader)
     }
 
     pub fn bytes_read(&self) -> usize {
         self.reader.bytes_read()
     }
 
+    pub fn stream_done(&mut self) {
```

Comment:
Is this intentionally `pub`? I expected the blocking implementation to handle this automatically.